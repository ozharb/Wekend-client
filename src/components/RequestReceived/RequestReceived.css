.Friends-List > h2{
    font-family: 'Public Sans', sans-serif;
    font-size: 50px;
   }
   .Friends-List > h3{
       font-family: 'Public Sans', sans-serif;
       color: yellow;
      }
      .friend-not-deleting-accepting{
      display: flex;
      justify-content: space-between;
      align-items: stretch;
       }
   .Friends-List_heading{
       font-size: 40px;
        margin: 0px;
   }
   .Friends-List:nth-child(even) {
       background: black;
       color: white;
   }
   .Friends-List:nth-child(even) p{
       color: white;
   }
   .Friends-List:nth-child(even) button{
    height: 100%;
       text-decoration: none;
       padding: 15px;
       background: white;
       color: black;
       width: 80px;
     
   }

   .Friends-List:nth-child(odd) {
       background: white;
       color: black;
   }
   .Friends-List:nth-child(odd) p{
       color: black;
   }
   .Friends-List:nth-child(odd) button{
       height: 100%;
       text-decoration: none;
       padding: 10px;
       background: black;
       color: white;
       width: 80px;
     
   }
 .delete-friend-yes{
     color: red;

 }
 .accept-friend-yes{
    color: green;

}

.yes-no-buttons{
    height: 100%;
    display: flex;
    flex-direction: column;
}
.yes-no-buttons>button{
     border: none; 
}
