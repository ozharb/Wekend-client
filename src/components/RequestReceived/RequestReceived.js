import toast, { Toaster } from 'react-hot-toast';
import React, { Component } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Link, withRouter } from 'react-router-dom'
import WekendApiService from '../../services/Wekend-api-service'

// import { Link } from 'react-router-dom'
// import { ThingStarRating } from '../ThingStarRating/ThingStarRating'
import './RequestReceived.css'
import FriendsContext from '../../contexts/FriendsContext';



export default class RequestReceived extends Component {
    static contextType = FriendsContext
    state = {
        expand: false,
        error: null,
        deleting: false,
        accepting: false
    };
    handleAccept = e => {
        this.setState({
            accepting: true,
            deleting: false
        })
    }
handleDelete=e=>{
    this.setState({deleting: true,
    accepting: false})
}
handleUnDelete=e=>{
    this.setState({deleting:false})
}
handleUnAccept=e=>{
    this.setState({accepting:false})
}
acceptFriendFinal = e => {
    e.preventDefault()
     
    let username = 'user'
    const user =  window.localStorage.getItem(username)
    const {friend} = this.props
    const friendId = friend.friend_id

const userId = friend.sender === user ? friend.sender_id : friend.receiver_id

  WekendApiService.acceptFriendRequest(friendId)
  .then(this.context.confirmFriend(friendId,userId))
  .catch(error => { console.error({ error }) })
  }
deleteFriendFinal = e => {
    e.preventDefault()
     
    let username = 'user'
    const user =  window.localStorage.getItem(username)
    const {friend} = this.props
    const friendId = friend.friend_id

const userId = friend.sender === user ? friend.sender_id : friend.receiver_id

  WekendApiService.deleteFriendship(friendId)
  .then(this.context.deleteFriend(friendId,userId))
  .catch(error => { console.error({ error }) })
  }
    render() {

        let username = 'user'
        const user = window.localStorage.getItem(username)
        const { friend } = this.props

        // const expandButtonText = this.state.expand
        // ?  <i className="fas fa-chevron-down"><FontAwesomeIcon className='chevron' icon='chevron-down' /></i> 
        // :   <i className="fas fa-chevron-right"><FontAwesomeIcon className='chevron' icon='chevron-right' /></i>

 const friendName = (friend) => {
     if(this.state.deleting) {
       return  `Delete ${friend}'s request?`
} else if (this.state.accepting){
    return `Accept ${friend}'s request?`
} else {
    return friend
}
     } 

        return (
            <>
                {/* <Link to={`/thing/${thing.id}`} className='ThingListItem'> */}

                <div className='Friends-List'>
                    <Toaster position="top" />
                    <div className='current-friend'>
                            <div className='friend-not-deleting'>
                            {this.state.accepting
                        ?<div className='yes-no-buttons'>
                        <button className='accept-friend' onClick={this.handleUnAccept}>No</button> 
                        <button className='accept-friend-yes' onClick={this.acceptFriendFinal}><span className='accept-friend-yes'>Yes</span></button>
                        </div>
                        : 
                        <button className='delete-friend' onClick={this.handleAccept}>Accept</button>
                        }
        <h2 className='Friends-List_heading' >{friendName(friend.friend)}</h2>
                        <div className='delete-friend-container'>
                            {this.state.deleting
                        ?<div className='yes-no-buttons'>
                        <button className='delete-friend' onClick={this.handleUnDelete}>No</button> 
                        <button className='delete-friend-yes' onClick={this.deleteFriendFinal}><span className='delete-friend-yes'>Yes</span></button>
                        </div>
                        : 
                        <button className='delete-friend' onClick={this.handleDelete}>Delete</button>
                        }
                        </div>
                        </div>
        
                    </div>
                </div>
                {/* </Link> */}
            </>
        )
    }
}

// function readableReviewCount(number) {
//   switch(number) {
//     case 0:
//       return 'no reviews yet'

//     case 1:
//       return `based on 1 review`

//     default:
//       return `based on ${number} reviews`
//   }
// }

// function truncate(text) {
//   const words = text.split(' ')

//   if (words.length > 10) {
//     return words.slice(0, 10).join(' ') + ' ...'
//   }

//   return text
// }
