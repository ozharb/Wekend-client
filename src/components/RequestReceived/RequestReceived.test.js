import React from 'react';
import RequestReceived from './RequestReceived';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

it('renders the UI as expected', () => {
  const tree = renderer
    
    .create(
    <BrowserRouter>
    <RequestReceived 
    props={{friend:{friend:''}}} friend={{friend:''}}
    />
     </BrowserRouter>
    )

    
    .toJSON();
 
  expect(tree).toMatchSnapshot();  
  });