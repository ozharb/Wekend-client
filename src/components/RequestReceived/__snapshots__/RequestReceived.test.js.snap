// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Friends-List"
>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "position": "fixed",
        "zIndex": 9999,
      }
    }
  />
  <div
    className="current-friend"
  >
    <div
      className="friend-not-deleting-accepting"
    >
      <div
        className="accept-friend-container"
      >
        <button
          className="accept-friend"
          onClick={[Function]}
        >
          accept
        </button>
      </div>
      <h2
        className="large-font"
      >
        
      </h2>
      <div
        className="delete-friend-container"
      >
        <button
          className="delete-friend"
          onClick={[Function]}
        >
          delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
